chore: Include a tool to migrate v1alpha5 manifests to v1beta1 ones (#4690)
aws/karpenter: tools/karpenter-convert/README.md

> # karpenter-converter

Reviewer: ```suggestion
# karpenter-convert
```

> karpenter-converter is a simple CLI tool to bump the API manifests from `v1alpha5` to `v1beta1`.

Reviewer: ```suggestion
karpenter-converter is a simple CLI tool to bump the API manifests from alpha to beta.
```

> It converts `Provisioner` to `NodePool` and `AWSNodeTemplate` to `EC2NodeClass`.

Reviewer: ```suggestion
It converts `v1alpha5/Provisioner` to `v1beta1/NodePool` and `v1alpha1/AWSNodeTemplate` to `v1beta1/EC2NodeClass`.
```

## Installation 

```

> go install github.com/aws/karpenter/tools/karpenter-converter/cmd/karpenter-converter

Reviewer: ```suggestion
go install github.com/aws/karpenter/tools/karpenter-convert/cmd/karpenter-convert
```

```

## Usage:

> ```

Reviewer: ```suggestion
```console
```

# Convert a single Provisioner file to NodePool
karpenter-convert -f provisioner.yaml > nodepool.yaml

# Convert a single AWSNodeTemplate file to EC2NodeClass
karpenter-convert -f nodetemplate.yaml > nodeclass.yaml

> # Convert an entire directory (.json, .yaml, .yml files) to the equivalent new APIs

Reviewer: Maybe a usage command to apply the new manifest to the cluster?

```console
karpenter-convert -f nodetemplate.yaml | kubectl apply -f -
```

# karpenter-convert

karpenter-convert is a simple CLI tool to bump the API manifests from alpha to beta.
It converts `v1alpha5/Provisioner` to `v1beta1/NodePool` and `v1alpha1/AWSNodeTemplate` to `v1beta1/EC2NodeClass`.

## Installation 

```

> go install github.com/aws/karpenter/tools/karpenter-convert/cmd/karpenter-convert

Reviewer: ```suggestion
go install github.com/aws/karpenter/tools/karpenter-convert/cmd/karpenter-convert@latest
```
